bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,248 @@
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogin() throws EtherRainException {
+        try {
+            sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password);
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogout() {
+        try {
+            sendGet("/ergetcfg.cgi?m=o");
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private synchronized List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return new LinkedList<String>();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return new LinkedList<String>();
+        }
+
+        return new BufferedReader(new StringReader(response.getContentAsString())).lines().collect(Collectors.toList());
+    }
+
+    private static List<EtherRainUdpResponse> updBroadcast() {
+
+        LinkedList<EtherRainUdpResponse> rList = new LinkedList<EtherRainUdpResponse>();
+
+        // Find the server using UDP broadcast
+
+        try (DatagramSocket c = new DatagramSocket()) {
+
+            c.setSoTimeout(BROADCAST_TIMEOUT);
+            c.setBroadcast(true);
+
+            byte[] sendData = BROADCAST_DISCOVERY_MESSAGE.getBytes();
+
+            // Try the 255.255.255.255 first
+            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length,
+                    InetAddress.getByName("255.255.255.255"), BROADCAST_DISCOVERY_PORT);
+            c.send(sendPacket);
+
+            while (true) {
+                // Wait for a response
+                byte[] recvBuf = new byte[15000];
+                DatagramPacket receivePacket;
+                try {
+                    receivePacket = new DatagramPacket(recvBuf, recvBuf.length);
+                    c.receive(receivePacket);
+                } catch (SocketTimeoutException e) {
+                    return rList;
+                }
+
+                // Check if the message is correct
+                String message = new String(receivePacket.getData()).trim();
```

Comment:
When creating a string from bytes you should specify the Charset it should use, otherwise it will use the default for the system which can vary based on locale.